zvariant/src/lib.rs

```diff
@@ -1293,6 +1293,158 @@ mod tests {
         assert_eq!(dict1, dict2);
     }
 
+    fn mock_fd<'a>(value: i32) -> Fd<'a> {
+        use std::os::fd::BorrowedFd;
+        use std::os::fd::RawFd;
+        let fd = unsafe { BorrowedFd::borrow_raw(value as RawFd) };
+        Fd::from(fd)
+    }
```

Comment:
You can just use `std::stdin().as_fd().into()` and then we don't need this function or at least it can be even simpler/safer?

Reply:
We cannot use `std::io::stdin().as_fd()` is because `std::io::stdin()` creates a temporary Stdin object (with a once-lock mutex internally) so it will be dropped without assignment.

Reply:
ah right but you can always [clone it](https://doc.rust-lang.org/std/os/fd/struct.BorrowedFd.html#method.try_clone_to_owned).

Reply:
Oh, sorry I didn't make it clear. The temporary value is the `Stdin` object but not the `BorrowedFd` Object. The Stdin object is not cloneable.

Even if the Stdin object is somehow resolved, I still don't suggest cloning the BorrowedFd object because cloning a BorrowedFd object requires an extra syscall `fcntl(fd, F_DUPFD)` and close the cloned FD on `Drop`, which, in my humble option, is unnessary for testing.

Reply:
> Even if the Stdin object is somehow resolved, I still don't suggest cloning the BorrowedFd object

I think it was me who as not clear I think cause I meant that you can create an `OwnedFd` from it (which is why I linked to `try_clone_to_owned`).

> Even if the Stdin object is somehow resolved, I still don't suggest cloning the BorrowedFd object because cloning a BorrowedFd object requires an extra syscall `fcntl(fd, F_DUPFD)` and close the cloned FD on `Drop`, which, in my humble option, is unnessary for testing.

The fact that it's a test, actually means we shouldn't care about the extra syscalls. I'd prefer less/simpler code in over optimizations in this context.

Reply:
`mock_fd()` is now removed since we migrate to use non-macro code in `dict_value()` test case.